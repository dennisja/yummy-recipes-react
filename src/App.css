.active-nav-link {
  border-bottom: 1px solid white;
}

#user-info {
  color: 'black';
  font-size: '12px';
}

#user-info td {
  padding: 3px;
}

#user-info tr {
  line-height: 40px;
  color: black;
}

.active-sidebar-link {
  background-color: #ff9800;
}

.active-nav-link:hover, .active-sidebar-link:hover, .nav-wrapper .right a:hover, .nav-wrapper .side-nav a:hover {
  background-color: white;
  color: #ff9800;
}


.collection a.collection-item{
  color: #ff9800;
}

@media all and (min-width: 993px){
  .nav-wrapper{
    padding-right: 20px;
    padding-left: 20px;
  }    
}

.btn-small{
  text-transform: capitalize;
  height: 30px;
  line-height: 30px;
}

.center-align .btn{
  margin-right: 5px;
}

hr{
  border: 1px solid #ff9800;
}

li.active .collapsible-header{
  text-align: -webkit-center;
  background-color: #ff9800;
  color: white;
}

#toast-container {
  top: auto !important;
  right: auto !important;
  bottom: 20%;
  left:40%;
}

input[type=text]:not(.browser-default):focus:not([readonly]),
input[type=search]:not(.browser-default):focus:not([readonly]),
input[type=email]:not(.browser-default):focus:not([readonly]),
input[type=password]:not(.browser-default):focus:not([readonly]),
textarea.materialize-textarea:focus:not([readonly]){
  border-bottom: 1px solid #ff9800;
  box-shadow: 0 1px 0 0 #ff9800;
  -webkit-box-shadow: 0 1px 0 0 #ff9800;
}

.recipe-card-holder .card-action .btn-small{
  padding-left: 6px;
  padding-right: 6px;
  margin-right: 5px;
}

.recipe-card-holder .card .card-action{
  padding: 8px 12px;
}

.recipe-card-holder .card .card-content {
  padding: 10px;
}

.recipe-card-holder .card .card-title{
  font-size: 18px;
}

.recipe-card-holder .card .card-reveal .card-title{
  font-size: 16px;
}

@media all and (max-width: 600px){
  #searchForm .input-field .btn{
    padding: 0 5px;
  }
}

#search-form-holder .card .card-content{
  padding-bottom: 0;
}