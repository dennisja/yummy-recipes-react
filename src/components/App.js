import axios from 'axios';
import { BrowserRouter } from 'react-router-dom';
import React, { Component } from 'react';

import Main from './Main';
import '../App.css';
import Header from './header/Header';
import Footer from './footer/Footer';
import Token from '../helpers/Token';


class App extends Component {
  static initialState = {
    loggedIn: false,
    userData: {
      data: null,
      token: "",
    },
  }
  
  state = App.initialState;

  loginUser = (userData) => {
    this.setState({
        loggedIn: true,
        userData: userData,
    })
    //put data in locationStorage
    Token.setToken(userData);
    // console.log(userData)
  }

  logoutUser = ()=>{
    this.setState(App.initialState);
    //clear data from location storage
    Token.deleteToken();
  }

  componentDidMount(){
    const userData = Token.getToken();
    if(userData){
      this.loginUser(userData);
    }
  }

  render() {
    const { loggedIn, userData } = this.state;
    return (
      <BrowserRouter>
        <div>          
          <Header 
            loggedIn={loggedIn} 
            logoutUser={this.logoutUser} 
            userData={userData.data}/>          
          <div className="container">
            <Main 
              loggedIn={loggedIn} 
              userData={userData.data} 
              loginUser={this.loginUser} />
          </div>
          <Footer isLoggedIn={loggedIn}/>
        </div>
      </BrowserRouter>
    );
  }
}

export default App;
