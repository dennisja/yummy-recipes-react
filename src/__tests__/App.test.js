import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import App from '../components/App';

describe('Test App top level component', () => {
  const appWrapper = shallow(<App />);
  it('renders without crashing', () => {
    expect(shallowToJson(appWrapper)).toMatchSnapshot();
  });

  it('has the right initial state', () => {
    expect(appWrapper.state().loggedIn).toEqual(false);
    expect(appWrapper.state().userData.data).toEqual(null);
    expect(appWrapper.state().userData.token).toEqual('');
  });

  it('renders a Header, Main and Footer components', () => {
    expect(appWrapper.find('Header')).toHaveLength(1);
    expect(appWrapper.find('Main')).toHaveLength(1);
    expect(appWrapper.find('Footer')).toHaveLength(1);
  });
});
