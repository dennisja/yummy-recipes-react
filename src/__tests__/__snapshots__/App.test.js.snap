// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App top level component renders without crashing 1`] = `
<BrowserRouter>
  <div>
    <Header
      loggedIn={false}
      logoutUser={[Function]}
      userData={null}
    />
    <div
      className="container"
    >
      <_class
        options={Object {}}
      />
      <Main
        loggedIn={false}
        loginUser={[Function]}
        updateUserData={[Function]}
        userData={null}
      />
    </div>
    <Footer
      isLoggedIn={false}
    />
  </div>
</BrowserRouter>
`;
